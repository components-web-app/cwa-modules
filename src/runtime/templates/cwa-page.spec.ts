import { describe, expect, test, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import * as nuxt from '#app'
import CwaPage from './cwa-page.vue'

describe('CWA page', () => {
  function createWrapper (iri = '12345') {
    // @ts-ignore
    vi.spyOn(nuxt, 'useNuxtApp').mockImplementation(() => ({
      $cwa: {
        resources: {
          pageIri: {
            value: iri
          }
        }
      }
    }))

    const result = mount(CwaPage, {
      global: {
        stubs: {
          ResourceLoader: true
        }
      }
    })

    console.log(result)

    return result
  }

  test('should display ResourceLoader component IF pageIri is defined', () => {
    const wrapper = createWrapper()

    expect(wrapper.findComponent({ name: 'ResourceLoader' })).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should NOT display ResourceLoader IF pageIri is NOT defined', () => {
    const wrapper = createWrapper('')

    expect(wrapper.element).toMatchSnapshot()
  })
})
