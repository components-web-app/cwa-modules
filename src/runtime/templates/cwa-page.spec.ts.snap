// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CWA page > snapshots > should NOT display ResourceLoader IF pageIri is NOT defined 1`] = `<!--v-if-->`;

exports[`CWA page > snapshots > should display ResourceLoader component IF pageIri is defined 1`] = `
<resourceloader-stub
  componentprefix="CwaPages"
  iri="12345"
/>
`;
